mlora/pipeline/function.py

```diff
@@ -0,0 +1,71 @@
+from mlora.pipeline.messages import PipeMessage, PipeMessageType
+from mlora.pipeline.transport import Transport
+from mlora.model.modelargs import MultiLoraBatchData
+
+import logging
+import torch
+
+
+class SendOperator(torch.autograd.Function):
+    # helper to reduce the activation memory
+    @staticmethod
+    def forward(ctx,
+                phony: torch.Tensor,
+                tensor_data: torch.Tensor,
+                transport: Transport,
+                msg_id: int,
+                input_args: MultiLoraBatchData):
+        assert isinstance(tensor_data, torch.Tensor)
+
+        msg = PipeMessage(src_=transport.worker_name,
+                          dst_=transport.next_worker_name,
+                          msg_type_=PipeMessageType.ACTIVATIONS,
+                          msg_id_=msg_id,
+                          tensor_data_=tensor_data,
+                          batch_data_=input_args)
+        transport.send_message(msg, False)
```

Comment:
any timeout issue? do we need to check this? with try catch?

Reply:
the torch RPC framework will retry